prov/rxm/src/rxm_cq.c

```diff
@@ -151,10 +151,11 @@ static int rxm_match_rma_iov(struct rxm_recv_entry *recv_entry,
 			return ret;
 
 		count = match_iov[i].count;
-		offset = match_iov[i].iov[count - 1].iov_len;
 
-		i++;
 		j += count - 1;
+		offset = (((count - 1) == 0) ? offset : 0) +
```

Comment:
This can happen at other indices of iovec as well not just zeroth index.

Btw, I think the current implementation to "arrange" local recv iovec to match the remote seems more complicated than necessary. I think the whole thing can be refactored. One unnecessary thing the current code does is setting the the length of the match_iov based on remote buffer length. I think this may not be needed at all. The only thing we need to keep track is the offset into the recv iovec.

How about something like the following:
```
match_iov[0] = recv_entry_rxm_iov;

for (i = 0, j = 1; i < rma_iov->count && j < match_iov->count; i++, j++) {
        ret = get_match_iov(&match_iov[j-1], rma_iov->len, &match_iov[j])
        if (ret)
                return ret;
}

get_match_iov(rxm_iov, len, match_rxm_iov)
{
        for (i = 0, j = 0; i < rxm_iov->count; i++, j++) {
                if (rxm_iov->iov[j].iov_len > len)
                        break;
                len -= rxm_iov->iov[j].iov_len;
        }
        if (i == rxm_iov->count)
                return -FI_ETOOSMALL;

        match_rxm_iov->iov.iov_base = rxm_iov->iov[j].iov_base + len;
        match_rxm_iov->iov.iov_len = rxm_iov->iov[j].iov_len - len;
        // copy rest of rxm_iov to match_rxm_iov
        return 0;
}
```

Reply:
> This can happen at other indices of iovec as well not just zeroth index

This is not for zeroth index. `count - 1 == 0` just means that user's I/O vector at `j` index isn't elapsed completely and should re-used


I'll look into your proposal tomorrow. Thanks

Reply:
> This is not for zeroth index. count - 1 == 0 just means that user's I/O vector at j index isn't elapsed completely and should re-used for the next iteration

Got it. Thanks for the clarification.